### DISCUSSION TITLE
How to detect that a scheduled entry has been published?

### DISCUSSION THREAD

**Discussion by goraxan** [craftcms/cms#10858] [Q&A] [answered]
I need to notify users when a new entry is published, if the entry is published (status = Live) from the begining I can detect it easily, by checking the `firstSave` attribute at the `Entry::EVENT_AFTER_SAVE` event and checking that the status is Live.
The problem comes when the entry is first saved schedule ti be published in the future (with a post date in the future, meaning it's saved with a `status = Pending`).

![image](https://user-images.githubusercontent.com/8337784/161541148-ab3be499-fca9-4e65-aac2-63aff9d48439.png)

When the entry is automatically published, I can't detect it. I've tried to use `$entry->isAttributeDirty('status')` in the `EVENT_AFTER_SAVE` event in order to see if that attribute changed but it looks like it's always false.

Any ideas?

**Comment by brandonkelly** [comment]
[This module](https://github.com/putyourlightson/craft-element-status-events) can be used to track the moment an entry goes live.

**Reply by goraxan** [reply]
I'll have a look, thanks!

**Comment by Jensderond** [comment]
Any idea how this can be accomplished with 5.x? Cannot seem to find any event and the module is outdated.

**Comment by brandonkelly** [comment, marked as answer]
Just [PR’d a native solution for this](https://github.com/craftcms/cms/pull/17024) for Craft 5.7: there’s a new `staticStatuses` config setting, which can be used to opt into having entry statuses stored statically and only updated on entry save; plus an `update-statuses` CLI command that can be hit with a Cron job, which updates any entries with out-of-date statuses, triggering `save` events and clearing caches in the process. (Plus a new `Entry::$oldStatus` property that can be checked from `save` event handlers to see if the status just changed.)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with that change.
